clustering/gaia_wrapper.py

```diff
@@ -104,21 +75,13 @@ def search_nearest_neighbors(self, sound_id, k, in_sound_ids=[], features='audio
         else:
             filter = None
         try:
-            if features == 'audio_as':
-                nearest_neighbors = self.as_view.nnSearch(sound_id, self.as_metric, filter).get(k)[1:]
-            elif features == 'tag':
-                nearest_neighbors = self.tag_view.nnSearch(sound_id, self.tag_metric, filter).get(k)[1:]
-            elif features == 'audio_fs':
-                nearest_neighbors = self.gaia_similiarity.view_pca.nnSearch(sound_id, 
-                                                                            self.gaia_similiarity.metrics['pca'], 
-                                                                            filter).get(k)[1:]
-            elif features == 'audio_fs_selected':
-                nearest_neighbors = self.fs_view.nnSearch(sound_id, self.fs_metric, filter).get(k)[1:]
-            elif features == 'audio_ac':
-                nearest_neighbors = self.ac_view.nnSearch(sound_id, self.ac_metric, filter).get(k)[1:]
+            gaia_view = getattr(self, '{}_view'.format(features))
+            gaia_metric = getattr(self, '{}_metric'.format(features))
+            nearest_neighbors = gaia_view.nnSearch(sound_id, gaia_metric, filter).get(k)[1:]
```

Comment:
The code of this file has improved A LOT :)
I guess the only thing missing is to treat `tag_features` just like the others :)